### PR TITLE
Always convert input to C-order in distance_transform_edt

### LINKED ISSUE #7660: distance_transform_edt not always working with cuCIM backend
**Describe the bug**
The distance_transform_edt returns all zero in special cases when cuCIM is installed and used as a backend. 
This happens when convert_to_cupy() decides that the data array needs to be copied [here:](https://github.com/Project-MONAI/MONAI/blob/16d4e2fcdbc6a7be5bf9df02fc12f34eee5f26b9/monai/transforms/utils.py#L2199)

**To Reproduce**
```
!pip install torch
!pip install cucim
!pip install monai
!pip install numpy

from monai.transforms import distance_transform_edt
import torch
import numpy as np

# create a c-contiguous array
mask = np.zeros((1, 4, 4))

# make it f-contiguous to force convert_to_cupy to copy the data array later on
mask2 = np.asfortranarray(mask)

# convert to torch tensor to make distance_transform_edt use cuCIM backend
mask = torch.tensor(mask, device='cuda')
mask2 = torch.tensor(mask2, device='cuda')

# set half of the mask to foreground
mask[0, 0:2, :,] = 1
mask2[0, 0:2, :,] = 1

# confirm masks are the same
print(mask)
print(mask2)

# calculate distance transform
edt = distance_transform_edt(mask)
edt2 = distance_transform_edt(mask2)

# check if the results are the same
print(edt)
print(edt2)
```
```
tensor([[[1., 1., 1., 1.],
         [1., 1., 1., 1.],
         [0., 0., 0., 0.],
         [0., 0., 0., 0.]]], device='cuda:0', dtype=torch.float64)
tensor([[[1., 1., 1., 1.],
         [1., 1., 1., 1.],
         [0., 0., 0., 0.],
         [0., 0., 0., 0.]]], device='cuda:0', dtype=torch.float64)
tensor([[[2., 2., 2., 2.],
         [1., 1., 1., 1.],
         [0., 0., 0., 0.],
         [0., 0., 0., 0.]]], device='cuda:0')
tensor([[[0., 0., 0., 0.],
         [0., 0., 0., 0.],
         [0., 0., 0., 0.],
         [0., 0., 0., 0.]]], device='cuda:0')
```

**Expected behavior**
Results should be the same, but the second output is all zero.

### PR DESCRIPTION
**Pull request by KumoLiu**
Fixes #7660

`indices_` may not always be a shallow copy for F-order input as it constantly converts to C-order
https://github.com/Project-MONAI/MONAI/blob/ffd4454b576abd4eaae30b364f41c213e30dca4c/monai/transforms/utils.py#L2209
https://github.com/Project-MONAI/MONAI/blob/ffd4454b576abd4eaae30b364f41c213e30dca4c/monai/utils/type_conversion.py#L262

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: monai/transforms/utils.py [hunk 1]
```diff
@@ -2190,7 +2190,7 @@ def distance_transform_edt(
         if return_distances:
             dtype = torch.float64 if float64_distances else torch.float32
             if distances is None:
-                distances = torch.zeros_like(img, dtype=dtype)  # type: ignore
+                distances = torch.zeros_like(img.contiguous(), dtype=dtype)  # type: ignore
```
Comment (ericspod) [inline monai/transforms/utils.py:2193]: This is going to make a copy of `img` if it isn't contiguous, instead I think we can use `torch.zeros_like(img, memory_format=torch.contiguous_format)`.

Comment (ericspod) [reply monai/transforms/utils.py]: Looks good to me, I'm not sure if the CI/CD issues are related however.

Comment (KumoLiu) [reply monai/transforms/utils.py]: Hi Eric, not related to this PR but for this one: https://github.com/Project-MONAI/MONAI/issues/7616.

**Review by ericspod — APPROVED** [review]

**Pull request merged**